yesod-test/Yesod/Test.hs

```diff
@@ -591,12 +605,57 @@ nameFromLabel label = do
 -- > <form method="POST">
 -- >   <label>Username <input name="f1"> </label>
 -- > </form>
+--
+-- Note that this function finds the labels in which contain the given word.
+-- It might choice labels unexpectedly or just fail in the circumstances like below:
+--
+-- > <form method="POST">
+-- >   <label for="nickname">Nickname</label>
+-- >   <input id="nickname" name="f1" />
+--
+-- >   <label for="nickname2">Nickname2</label>
+-- >   <input id="nickname2" name="f2" />
+-- > </form>
+--
+-- > request $ do
+-- >   byLabel "Nickname" "Snoyberger"
+--
+-- Therefore, this function is deprecated. Please consider using 'byLabelExact',
+-- which performs the exact match over the given word.
```

Comment:
I think we should use a GHC [`DEPRECATED` pragma](https://downloads.haskell.org/~ghc/7.0.3/docs/html/users_guide/pragmas.html), and recommend `byLabelWithMatch Data.Text.contains` for the old behavior and `byLabelExact` for the new behavior

Reply:
> recommend byLabelWithMatch Data.Text.contains for the old behavior and byLabelExact for the new behavior

Actually, `byLabelWithMatch` is the abstraction of the matching methods and `byLabel` performs the old behavior (contain) as before, unless users will be forced to rewrite their tests to use `byLabelContain` or whatever. And yes, `byLabelExact` is for the new behavior.

So,

`byLabel` -> deprecated
`byLabelExact` -> recommanded

Is this okay?

> I think we should use a GHC DEPRECATED pragma

NiceXD. I even didn't know that. I'll modify the code.